### PR TITLE
Add updated iOS App support to the Vulkan examples

### PR DESCRIPTION
**Pull request by SRSaunders**
This PR depends on and includes #1117 due to those dependencies - only the final commits here are new.  However, I am submitting this as a separate PR since I am not sure if you want to include iOS in your project.  I would be happy to provide support from time-to-time for new updates if you need it. I would be interested in @billhollings reaction and if he would like this added to your repository, or if he has other plans. Most of the original code was his, and credits/copyright remain in place but perhaps dates should be updated.

Similar to before, it is an Xcode project contained within a separate sub-folder (but named _apple_ this time) that depends on the base project.  It is like the previous xcode sub-folder, but with documentation and functionality emphasis on _iOS_ and the _iOS Simulator_.  _macOS_ standalone app support is also present but not emphasized.  Most users should use the base cmake project in the main folder for _macOS_.

Details are:

1. Updated to use **VK_USE_PLATFORM_METAL_EXT** by default vs. deprecated **VK_USE_PLATFORM_IOS_MVK**
2. Updated to use **MoltenVK.xcframework** from the Vulkan SDK (requires SDK 1.3.275.0 and Xcode 14 or later).
3. Added support for single and multi-touch gestures on _iOS_ and the _iOS Simulator_, allowing select and rotation of the scene, plus translation and zoom.  Virtual keyboard support remains for imgui overlay toggle and animation pausing.
4. Updated example.h file with categories and comments on which examples currently work and don't work on iOS.
5. Updated to work with your current main project baseline.
6. Updated docs.

Let me know if you are interested in this and if so, I will do my best to help out.  If not, then no worries.  ~~I realize that automated CI support is not implemented at this time, but perhaps could be added.~~

Here is a pic of the bloom example running on the iOS simulator...
Screenshot 2024-04-21 at 1 26 16 PM

### CODE REVIEW

**Comment by SaschaWillems** [comment]
Thank you very much for your PR. This is very much appreciated. I don't own any apple devices to test this on, so all I can do is a code review.

I will take a look at this, but it may take a few weeks.

**Comment by SRSaunders** [comment]
Again thanks for agreeing to look at this.  Only the final commit (https://github.com/SaschaWillems/Vulkan/pull/1119/commits/e4a42420a21ba1424e979b84827be65754cfbd9f) is different from the other PR.

Given your comments above about testing on Apple devices, I may go ahead and add CI support so you can be assured of build sanity.

**Comment by SRSaunders** [comment]
Fixed merge conflicts.

Thanks for catching my redundant code for getting `rayTracingPipelineProperties`.  I missed the common code for getting this in **VulkanRayTracingSample**.  Note as part of the merge I removed my unneeded calls to `vkGetPhysicalDeviceProperties2()` in the ray tracing examples.

**Review by SaschaWillems — CHANGES_REQUESTED** [review]
While I can't test this looks fine to me and CI is in place. Aside from a minor nitpick I'll merge once I get green light from @billhollings with regards to copyright.

File: benchmark-all-validate.py
Comment (SRSaunders) [reply benchmark-all-validate.py:1]: Done...

File: apple/README_MoltenVK_Examples.md
Comment (SRSaunders) [reply apple/README_MoltenVK_Examples.md:7]: Makes sense.  And as I mentioned before, if @billhollings has concerns about maintaining it I can help out.

Comment (billhollings) [reply apple/README_MoltenVK_Examples.md:7]: Sorry for the delay. 

Yes...any code Brenwill contributes (or has contributed) is open-source, and respects the open-source license of this repo.

**Review by billhollings — APPROVED** [review]

**Comment by SaschaWillems** [comment]
Thank you very much for the confirmation. Merging this now :)

**Pull request merged**
